### PR TITLE
Stack Recipe Lists

### PR DESCRIPTION
**Pull request by ghost**
Ports
https://github.com/Baystation12/Baystation12/pull/3143
And some later changes

Basically, when you use metal in hand to bring up the recipes, similar recipes can be put into a stack_recipe_list, so the recipe menu is not spammed with similar recipes. You click the recipe (example: "office chairs", and that brings up another menu with more recipes "dark office chair (5 sheets)" to make

### CODE REVIEW

File: code/game/objects/items/stacks/stack.dm [hunk 1]
```diff
@@ -86,46 +87,61 @@
...
-			var/list/multipliers = list(5,10,25)
-			for (var/n in multipliers)
-				if (max_multiplier>=n)
-					t1 += " <A href='?src=[REF(src)];make=[i];multiplier=[n]'>[n*R.res_amount]x</A>"
-			if (!(max_multiplier in multipliers))
-				t1 += " <A href='?src=[REF(src)];make=[i];multiplier=[max_multiplier]'>[max_multiplier*R.res_amount]x</A>"
+
+		if (istype(E, /datum/stack_recipe_list))
+			var/datum/stack_recipe_list/srl = E
+			if (src.amount >= srl.req_amount)
```
Comment (Jalleo) [inline code/game/objects/items/stacks/stack.dm]: is the src needed in these datums?

**Review by Jalleo — COMMENTED** [review]
Question

File: code/game/objects/items/stacks/stack.dm (hunk 1 above)
Comment (uraniummeltdown) [reply code/game/objects/items/stacks/stack.dm]: Probably not but it doesn't break anything if you leave the src. in, I'll clean up the src. once it works
The issue is that in all stack recipe menus clicking the 'links' does nothing it does not produce the item

**Comment by optimumtact** [comment]
Next time, get someone willing to work with you before wasting our time looking at your pr

**Comment by uraniummeltdown** [comment]
its fixed now you dick

**Pull request closed without merging**